schemas/stsci.edu/asdf/0.1.0/core/column.yaml

```diff
@@ -0,0 +1,49 @@
+%YAML 1.1
+---
+$schema: "http://stsci.edu/schemas/yaml-schema/draft-01"
+id: "http://stsci.edu/schemas/asdf/0.1.0/core/column"
+tag: "tag:stsci.edu:asdf/0.1.0/core/column"
+
+title: >
+  A column in a table.
+
+description: |
+  Each column contains a name and an array of data, and an optional description
+  and unit.
+
+type: object
+properties:
+  name:
+    description: |
+      The name of the column.  Each name in a
+      [table](http://stsci.edu/schemas/asdf/0.1.0/core/table) must be
+      unique.
+    type: string
+    pattern: "[A-Za-z_][A-Za-z0-9_]*"
+
+  data:
+    description: |
+      The array data for the column.
+    allOf:
+      - $ref: ndarray
+
+  description:
+    description: |
+      An optional description of the column.
+    type: string
+    default: ''
+
+  unit:
```

Comment:
Do you want a `format` key as well?  This is maybe a philosophical question of whether a column should have an intrinsic property of how to best represent itself to the user.  I like this idea, but I'll admit that in practice I very rarely set the column format in my day to day work.